### PR TITLE
add tests for all the methods in class UnsignedByteArray

### PR DESCRIPTION
**Pull request by mukulljangid**
Addresses: https://github.com/XRPLF/xrpl4j/issues/248

### CODE REVIEW

File: xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java [hunk 1]
```diff
@@ -41,4 +42,96 @@ public void lowerCaseOrUpperCase() {
     assertThat(UnsignedByteArray.fromHex("abcdef0123").hexValue()).isEqualTo("ABCDEF0123");
   }
 
+  @Test
+  public void simpleEmpty() {
+    assertThat(UnsignedByteArray.of(new byte[] {0}).empty()).isEqualTo(UnsignedByteArray.of(new byte[] {}));
+    assertThat(UnsignedByteArray.of(new byte[] {0, MAX_BYTE}).empty()).isEqualTo(UnsignedByteArray.of(new byte[] {}));
```
Comment (nkramer44) [inline xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java:48]: Why do we need these? `empty()` is a static method, so it's a bit weird to call `empty()` on an instance of `UnsignedByteArray`. Also, `empty()` just creates a new empty `UnsignedByteArray`, so why create a new `UnsignedByteArray` and then call `empty()`?

File: xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java [hunk 2]
```diff
@@ -41,4 +42,96 @@ public void lowerCaseOrUpperCase() {
...
+    assertThat(UnsignedByteArray.of(new byte[] {MAX_BYTE, MAX_BYTE}).empty().length()).isEqualTo(0);
+    assertThat(
+      UnsignedByteArray.of(new byte[] {0, MAX_BYTE}).empty().equals(UnsignedByteArray.of(new byte[] {}))
+    ).isTrue();
+  }
+
+  @Test
+  public void simpleLength() {
```
Comment (nkramer44) [inline xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java:56]: can we get rid of the "simple" prefix on all of these names? The fact that there is no "complicated" counterpart is a bit confusing lol

**Review by nkramer44 — CHANGES_REQUESTED** [review]
Can you add the following tests as well?
1) One for `toByteArray()`
2) A test that calls `.equals()` on the same instance, like `assertThat(unsignedByteArray).isEqualTo(unsignedByteArray)`
3) A test that calls `.equals()` on an object of a different type. Something like
`assertThat(unsignedByteArray).isNotEqualTo(new Object())`

**Review by sappenin — CHANGES_REQUESTED** [review]

File: xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java [hunk 3]
```diff
@@ -150,7 +162,7 @@ public void simpleHashcode() {
   }
 
   @Test
-  public void simpleToString() {
+  public void UnsignedByteArrayToString() {
```
Comment (sappenin) [inline xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java:165]:
```suggestion
  public void unsignedByteArrayToString() {
```

File: xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java (hunk 1 above)
Comment (sappenin) [reply xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java]: @nkramer44 I think this is now resolved with the latest commits from @mukulljangid - plz make another comment if you see something missing.

**Review by sappenin — APPROVED** [review]

**Review by nkramer44 — CHANGES_REQUESTED** [review]
Small nit -- Can you add a test that calls `equals` with a non UnsignedByteArray type, like `assertThat(array.equals(new Object()).isFalse()`?

**Review by nkramer44 — APPROVED** [review]

**Pull request merged**